nca/Parsers/IstioGatewayPolicyGenerator.py

```diff
@@ -77,6 +82,22 @@ def create_istio_gateway_policies(self):
         if unused_gateways:
             self.vs_parser.warning(f'the following gateways have no virtual services attached: '
                                    f'{",".join([gtw.full_name() for gtw in unused_gateways])}')
+        result = self.ingress_gtw_to_mesh_policies.copy()
+        if self.mesh_to_egress_gtw_policies and self.egress_gtw_to_dns_policies:
+            # only if we have both parts of egress flow, will we model the whole egress flow (including deny policies)
+            # Note: we assume here that there is a single egress flow (and the 'hosts' and other attributes
+            # match between mesh-to-egress and egress-to-dns parts of the flow).
+            # A more refined check may be done, that checks those attributes.
+            result.extend(self.mesh_to_egress_gtw_policies)
+            result.extend(self.egress_gtw_to_dns_policies)
+            assert self.deny_mesh_to_dns_policies
+            result.extend(self.deny_mesh_to_dns_policies)
+        elif self.mesh_to_egress_gtw_policies or self.egress_gtw_to_dns_policies:
+            part1_status = "specified" if self.mesh_to_egress_gtw_policies else "missing"
+            part2_status = "specified" if self.egress_gtw_to_dns_policies else "missing"
+            self.vs_parser.warning(f"Ignoring partially defined egress flow: mesh-to-egress-gateway part of the flow "
+                                   f"is {part1_status}, while egress-gateway-to-external-service part of the flow "
+                                   f"is {part2_status}")
```

Comment:
rephrase the warning, add details on the VS ignored

Reply:
Done